sprokit/processes/ocv/detect_in_subregions_process.cxx

```diff
@@ -0,0 +1,377 @@
+      {
+        continue;
+      }
+
+      int x = bbox.upper_left()[0];
+      int y = bbox.upper_left()[1];
+      int w = bbox.width();
+      int h = bbox.height();
+
+      if( m_method == FIXED_SIZE )
+      {
+        int cx = ( x + w ) / 2;
+        int cy = ( y + h ) / 2;
+
+        bool intersect_found = false;
+
+        for( auto region : previous_regions )
+        {
+          if( region.contains( cv::Point( cx, cy ) ) )
+          {
+            intersect_found = true;
+            break;
+          }
+        }
+        if( intersect_found )
+        {
+          continue;
+        }
+
+        x = std::max( 0, cx - m_fixed_size / 2 );
+        y = std::max( 0, cy - m_fixed_size / 2 );
+
+        w = m_fixed_size;
+        h = m_fixed_size;
+
+        if( x + w > img.width() )
+        {
+          w = img.width() - x;
+        }
+        if( y + h > img.height() )
+        {
+          h = img.height() - y;
+        }
+        if( w <= 1 || h <= 1 )
+        {
+          continue;
+        }
+      }
+
+      LOG_TRACE( m_logger, "Processing ROI window with upper left coordinates (" +
+                 std::to_string(x) + "," + std::to_string(y) + ") of size (" +
+                 std::to_string(w) + "x" + std::to_string(h) + ")" );
+
+      // TODO: ocv is only used to crop the image. This can be replaced by a
+      // vital image cropping utility once that becomes available, and then this
+      // can be moved to a core process.
+
+      // Make CV rect for bbox
+      cv::Rect roi( x, y, w, h );
+
+      // Detect within the region of interest.
+      windowed_image = vital::image_container_sptr(
+        new arrows::ocv::image_container(
+          cv_src( roi ), arrows::ocv::image_container::BGR_COLOR ) );
+
+      auto dets = m_detector->detect( windowed_image );
+
+      for( auto det : *dets )
+      {
+        auto det_bbox = det->bounding_box();
+        vital::bounding_box_d::vector_type offset( 2 );
+        offset[0] = x; offset[1] = y;
+        kwiver::vital::translate( det_bbox, offset );
+        det->set_bounding_box( det_bbox );
+      }
+
+      // Add detections set to the output detection set
+      dets_out.insert( dets_out.end(), dets->cbegin(), dets->cend() );
+
+      timer.stop();
+
+      processed_count++;
+      previous_regions.push_back( roi );
+
+      LOG_TRACE( m_logger, "Time to classify window: " << timer.elapsed() );
+    } // end foreach
+
+    dets_out_sptr = std::make_shared<vital::detected_object_set>( dets_out );
+  }
+
+}; // end priv class
+
+
+// =============================================================================
+detect_in_subregions_process
+::detect_in_subregions_process( kwiver::vital::config_block_sptr const& config )
+  : process( config ),
+    d( new detect_in_subregions_process::priv )
+{
+  // Attach our logger name to process logger
+  attach_logger( kwiver::vital::get_logger( name() ) ); // could use a better approach
```

Comment:
I think this logger naming is now done by the process base class and does not need to be done here.